modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/ElasticsearchIndex.java

```diff
@@ -839,4 +839,27 @@ private SearchResult<IndexTheme> getByQuery(ThemeSearchQuery query, int maxRetry
       throw new SearchIndexException("Error querying theme index", t);
     }
   }
+
+  /**
+   * Escapes all reserved Elasticsearch characters in a given string
+   * Useful for when a given query string does not know about Elasticsearch query syntax
+   * @param text String to escape reserved characters in
+   * @return the given string with escaped characters
+   */
+  public String escapeQuery(String text) {
+    StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < text.length(); i++) {
+      char c = text.charAt(i);
+      // These characters are part of the query syntax and must be escaped
+      // CHECKSTYLE:OFF
+      if (c == '\\' || c == '+' || c == '-' || c == '!' || c == '(' || c == ')' || c == ':'
+              || c == '^' || c == '[' || c == ']' || c == '\"' || c == '{' || c == '}' || c == '~'
+              || c == '*' || c == '?' || c == '|' || c == '&' || c == '/') {
+        sb.append('\\');
+      }
+      // CHECKSTYLE:ON
+      sb.append(c);
+    }
+    return sb.toString();
+  }
```

Comment:
```suggestion
public String escapeQuery(String text) {
    int length = text.length();
    Set<Character> specialCharacters = new HashSet<>(Arrays.asList('\\', '+', '-', '!', '(', ')', ':', '^',
            '[', ']', '\"', '{', '}', '~', '*', '?', '|', '&', '/'));

    StringBuilder sb = new StringBuilder(length + (length / 10));
    for (char c : text.toCharArray()) {
        if (specialCharacters.contains(c)) {
            sb.append('\\');
        }
        sb.append(c);
    }
    return sb.toString();
}
```

Reply:
Or just `Set.of(...)` instead of `new HashSet<>(Arrays.asList(...))` :D